### PR TITLE
Add cspell workflow

### PR DESCRIPTION
**Pull request by miles-six**
@janjakubnanista  I pull the latest code and try again #479

### CODE REVIEW

**Comment by janjakubnanista** [comment]
@miles-six nice, thanks! Could you pls try rebasing again, I can't seem to be able to approve the workflow runs and github is not great at telling me why, it might be that the branch is just outdated

**Comment by miles-six** [comment]
@janjakubnanista   
thanks sir, can you check it again?

File: .github/workflows/reusable-test.yaml [hunk 1]
```diff
@@ -147,3 +147,19 @@ jobs:
...
+      uses: actions/checkout@v4
+
+    - name: Setup environment
+      uses: ./.github/workflows/actions/setup-environment
+
+    - name: Install dependencies
+      uses: ./.github/workflows/actions/install-dependencies
+
+    - name: Spell Check
+      run: pnpm cspell --config ./cspell.json
```
Comment (janjakubnanista) [inline .github/workflows/reusable-test.yaml:165]: It looks like the indentation here is not correct, breaking the workflow https://github.com/LayerZero-Labs/devtools/actions/runs/8542631863

**Review by janjakubnanista — CHANGES_REQUESTED** [review]

**Comment by miles-six** [comment]
@janjakubnanista  resolved it

**Comment by miles-six** [comment]
@janjakubnanista  I can run **cspell.** local successfully.I don't know why failed on github

**Comment by janjakubnanista** [comment]
@miles-six I'll investigate the error in the CI - it's some real weirdness since `pnpm install` is trying to clone git repositories for some reason

**Comment by janjakubnanista** [comment]
For some reason I was getting permission denied and don't have that much time to investigate - what I did was I pushed into [this branch](https://github.com/LayerZero-Labs/devtools/tree/miles-six/main) - I can either recreate your PR or you can fetch & hard reset your branch to this one then all should be good

**Comment by janjakubnanista** [comment]
There still seems to be an issue with the command - I think it might need to be changed to `cspell lint`:

Screenshot 2024-04-04 at 11 59 37 AM

**Comment by miles-six** [comment]
> For some reason I was getting permission denied and don't have that much time to investigate - what I did was I pushed into [this branch](https://github.com/LayerZero-Labs/devtools/tree/miles-six/main) - I can either recreate your PR or you can fetch & hard reset your branch to this one then all should be good

Ok sir.Just do it your way. Thanks

**Comment by miles-six** [comment]
@janjakubnanista  I change to cspell lint .

**Comment by janjakubnanista** [comment]
@miles-six I have some high-priority things that I need to tend to, I will come back to you on this one in a bit

**Comment by miles-six** [comment]
Yes. Waiting for it

**Comment by hadifalahi4142** [comment]
this project can a grow now

**Comment by TRileySchwarz** [comment]
Closed for now, will reprioritize this as needed.

**Pull request closed without merging**